streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedStreamImpl.java

```diff
@@ -58,25 +59,44 @@ public KGroupedStreamImpl(final KStreamBuilder topology,
                                final String storeName) {
         Objects.requireNonNull(reducer, "reducer can't be null");
```

Comment:
If the check was removed the unit tests would fail.  I'm not overly concerned either way, but my preference is to remove unnecessary duplication. I get you point about the public API and i generally agree, but in this case it is not like the checks are buried multiple levels down. They have just been moved to the overloaded function that is immediately called.